content/en/docs/reference/kubectl/conventions.md

```diff
@@ -36,42 +36,56 @@ For `kubectl run` to satisfy infrastructure as code:
-| CronJob (deprecated)                | batch/v1beta1      | `kubectl run --generator=cronjob/v1beta1`         |
-| CronJob (deprecated)                | batch/v2alpha1     | `kubectl run --generator=cronjob/v2alpha1`        |
+{{< table caption="Resources you can create using kubectl run" >}}
+| Resource                             | API group          | kubectl command                                   |
+|--------------------------------------|--------------------|---------------------------------------------------|
+| Pod                                  | v1                 | `kubectl run --generator=run-pod/v1`              |
+| ReplicationController _(deprecated)_ | v1                 | `kubectl run --generator=run/v1`                  |
+| Deployment _(deprecated)_            | extensions/v1beta1 | `kubectl run --generator=deployment/v1beta1`      |
+| Deployment _(deprecated)_            | apps/v1beta1       | `kubectl run --generator=deployment/apps.v1beta1` |
+| Job _(deprecated)_                   | batch/v1           | `kubectl run --generator=job/v1`                  |
+| CronJob _(deprecated)_               | batch/v2alpha1     | `kubectl run --generator=cronjob/v2alpha1`        |
+| CronJob _(deprecated)_               | batch/v1beta1      | `kubectl run --generator=cronjob/v1beta1`         |
+{{< /table >}}
+
 
 {{< note >}}
 Generators other than `run-pod/v1` are deprecated.
 {{< /note >}}
 
-If you do not specify a generator flag, other flags prompt you to use a specific generator. The following table lists the flags that force you to use specific generators, depending on the version of the cluster:
+If you do not explicitly specify a generator when you invoke `kubectl run`, a generator can be implicitly set based on the flags you chose. The following table lists flags and the generators that are activated:
 
+{{< table caption="kubectl run flags and the resource they imply" >}}
 |   Generated Resource  | Flag                    |
 |:---------------------:|-------------------------|
 | CronJob               | `--schedule=<schedule>` |
 | Deployment            | `--restart=Always`      |
 | Job                   | `--restart=OnFailure`   |
 | Pod                   | `--restart=Never`       |
+{{< /table >}}
 
 {{< note >}}
 These flags use a default generator only when you have not specified any flag.
-This means that when you combine `--generator` with other flags the generator that you specified later does not change. For example, in a cluster v1.4, if you initially specify
-`--restart=Always`, a Deployment is created; if you later specify `--restart=Always`
-and `--generator=run/v1`, a ReplicationController is created.
-This enables you to pin to a specific behavior with the generator,
-even when the default generator is changed later.
+This means that when you combine `--generator` with other flags, the generator that you specified later does not change.
```

Comment:
nit: clarify in another PR? This part still seems confusing to me -- default generator vs generated resource.
What happens to the generator -- it is not changed by using or overlaying a number of flag combinations.

Reply:
I had another go.

If this doesn't land via the v1.17 release, I can cherry-pick the same changes against _master_.